### DISCUSSION TITLE
MRTKv3 equivalent to NearInteractionTouchableUnityUI ?

### DISCUSSION THREAD

**Discussion by phwyfve** [microsoft/MixedRealityToolkit-Unity#10645] [Q&A] [answered]
I went through the MRTKv3 samples, but I couldn't find any interactions with raw unity ugui components. All buttons/sliders are MRTK specific ones (PressableButton, ..). The previous way to do that in MRTK v2 was to add a NearInteractionTouchableUnityUI component to the canvas. How can this be done with mrtkv3 ?

**Comment by Zee2** [comment, marked as answer]
Hi Jean;

The new Canvas-based `PressableButtons` (see `CanvasExample` and `CanvasUITearsheet`) are a replacement for your UnityUI needs; the buttons are fully compatible with UntiyUI input, masking, layouts, and all of the functionality you expect. We discourage folks from using vanilla UnityUI components in MR due to a lack of depth cues, interactivity, and fidelity; the new Canvas-based controls offer the best of both worlds, with UnityUI-compatible layout and editing workflow, as well as full volumetric feedback and usability.

If you find any issues with the Canvas-based UX controls, or any feature gap between them and what you expect from typical UnityUI setups, please file a bug!

(In the future, we may support primitive "touchability" on vanilla Canvas UI, as the XRI team at Unity has indicated this is one of their potential future interests. However, even if/when this happens, we will still strongly recommend using our full 3D controls, as they are much more usable, accessible, and intuitive for MR applications and contexts.)
